druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -1102,24 +1107,46 @@ public String getQueryString(String pagingIdentifier, int offset) {
         }
       }
       this.timeZone = timeZone;
+      this.dateFormatter = new SimpleDateFormat(TimeExtractionFunction.ISO_TIME_FORMAT,
+          Locale.ROOT);
+      if (timeZone != null) {
+        this.dateFormatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+      }
     }
 
     protected void clearFieldNameLists() {
       dimensions.clear();
       metrics.clear();
     }
 
-    @SuppressWarnings("incomplete-switch") String translate(RexNode e, boolean set) {
+    @SuppressWarnings("incomplete-switch")
+    String translate(RexNode e, boolean set, boolean formatDateString) {
       int index = -1;
       switch (e.getKind()) {
       case INPUT_REF:
         final RexInputRef ref = (RexInputRef) e;
         index = ref.getIndex();
         break;
       case CAST:
-        return tr(e, 0, set);
+        return tr(e, 0, set, formatDateString);
       case LITERAL:
-        return ((RexLiteral) e).getValue3().toString();
+
+        if (!formatDateString) {
+          return ((RexLiteral) e).getValue3().toString();
+        } else {
+          switch (((RexLiteral) e).getTypeName()) {
+          case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+          case TIMESTAMP:
+          case DATE:
+            //case CHAR:
+            // Case when we are passing to druid as an extractionFunction
+            // Need to format the timestamp Strng in druid format.
+            Long val = ((RexLiteral) e).getValueAs(Long.class);
+            return dateFormatter.format(val);
+          default:
+            return ((RexLiteral) e).getValue3().toString();
```

Comment:
looking at `getValue3` it can return nulls, probably it is better to use something like `Objects.toString`

Reply:
done.